test/integration/a_visitor_can_view_items_by_category_test.rb

```diff
@@ -1,48 +1,50 @@
 require 'test_helper'
 
+
 class AVisitorCanViewItemsByCategoryTest < ActionDispatch::IntegrationTest
+  include CategoryItemsSetup
+
   test 'a visitor can visit category page to see all items in category' do
+    create_categories_and_items
 
-    category     = Category.create(title: "Snowboards")
-    category_two = Category.create(title: "Apparel")
-
-    Item.create(name: "gnar possum",
-                       description: "a snowboard for shredding gnar pow",
-                       price: 1000,
-                       category_id: category.id)
-    Item.create(name: "gwar possum",
-                       description: "a snowboard for gwar concerts",
-                       price: 15,
-                       category_id: category.id)
-    Item.create(name: "Sweet Jacket",
-                       description: "Keeps you warm",
-                       price: 240,
-                       category_id: category_two.id)
-    Item.create(name: "Hoody",
-                       description: "Keeps you slightly warm",
-                       price: 80,
-                       category_id: category_two.id)
-
-    visit category_path(category.slug)
+    visit category_path(@category.slug)
 
     assert page.has_content?("Snowboards")
-    assert page.has_content?("Name: GNAR POSSUM")
-    assert page.has_content?("Description: A snowboard for shredding gnar pow")
-    assert page.has_content?("Price: $1000")
-    assert page.has_content?("Name: GWAR POSSUM")
-    assert page.has_content?("Description: A snowboard for gwar concerts")
-    assert page.has_content?("Price: $15")
-
 
-    visit category_path(category_two.slug)
+    within('.items') do
+      assert page.has_content?("Name: GNAR POSSUM")
+      assert page.has_content?("Description: A snowboard for shredding gnar pow")
+      assert page.has_content?("Price: $1000")
+      assert page.has_content?("Name: GWAR POSSUM")
+      assert page.has_content?("Description: A snowboard for gwar concerts")
+      assert page.has_content?("Price: $15")
+      refute page.has_content?("Apparel")
+      refute page.has_content?("Name: SWEET JACKET")
+      refute page.has_content?("Keeps you warm")
+      refute page.has_content?("Price: $240")
+      refute page.has_content?("Name: HOODY")
+      refute page.has_content?("Keeps you slightly warm")
+      refute page.has_content?("Price: $80")
+    end
+
+    visit category_path(@category_two.slug)
 
     assert page.has_content?("Apparel")
-    assert page.has_content?("Name: SWEET JACKET")
-    assert page.has_content?("Keeps you warm")
-    assert page.has_content?("Price: $240")
-    assert page.has_content?("Name: HOODY")
-    assert page.has_content?("Keeps you slightly warm")
-    assert page.has_content?("Price: $80")
-  end
 
+    within('.items') do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.